v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -123,14 +123,38 @@ function formatSingleError(
   if (!message.includes(error.message)) {
     message = error.message;
   }
+
+  // We remove the node:assert and node:test error codes
   message = message
     .replace(" [ERR_ASSERTION]", "")
     .replace(" [ERR_TEST_FAILURE]", "");
 
   const diff = getErrorDiff(error);
 
   if (diff !== undefined) {
-    message = message.match(/^(AssertionError:\s.*)\:/)?.[1] ?? message;
+    // If we got a diff, we try to remove any diff from the message, which can
+    // have different formats.
+
+    // Format 1: A line starting with "+ actual - expected".
+    let match = message.match(/^(.*?)\n\s*?\+\sactual\s-\sexpected/s);
+
+    // Format 2: A line starting with "{error.actual} !==".
+    if (match === null && "actual" in error) {
+      match = message.match(/^(.*?)\n\s*?(?:.*?) !==/s);
```

Comment:
In https://github.com/NomicFoundation/hardhat/pull/6199/commits/9b0d5b19d37346739c84398b1de76d8b24c4a9db, I updated the first matching group to be greedy. That way we match everything up to the line with `!==`. I also updated the tests to catch if that was not the case.

Reply:
Looks good.